core/fetch_token.go

```diff
@@ -0,0 +1,87 @@
+package core
+
+import (
+	"net/http"
+	"net/url"
+)
+
+type FetchTokenByAuthorizationCodeOptions struct {
+	tokenEndpoint string
+	code          string
+	codeVerifier  string
+	clientId      string
+	redirectUri   string
+	resource      string
+}
+
+func FetchTokenByAuthorizationCode(client *http.Client, options *FetchTokenByAuthorizationCodeOptions) (CodeTokenResponse, error) {
+	values := url.Values{
+		"client_id":     {options.clientId},
+		"redirect_uri":  {options.redirectUri},
+		"code_verifier": {options.codeVerifier},
+		"code":          {options.code},
+		"grant_type":    {"authorization_code"},
+	}
+
+	if options.resource != "" {
+		values.Add("resource", options.resource)
+	}
+
+	response, requestErr := client.PostForm(options.tokenEndpoint, values)
```

Comment:
Can we use `requestError` instead of `requestErr`?

Reply:
Yes, we can.
But in Golang, we often use `err` to represent `error`, because the `error` is a built-in interface type. Also, Golang has a rule that all error constants should start with `Err`, like `ErrRequest`. so I intend to make them align.